Keto-Eating/Keto-Recipe-API
# Favorites

Wheww.... okay that was a long day of coding and I got stuck SOOO many times! 

I'm raising this PR because I'm planning on taking a break for the rest of the night (but we'll see, I might push some more changes later if I get a surge of inspiration)

**What I accomplished:**
- [x] Created Favorites model
- [x] Added "Heart" on recipe cards so we can save a recipe to favorites 
![image](https://user-images.githubusercontent.com/6496862/46900833-fc00ee80-ce5d-11e8-94bd-443d8cbcf5c5.png)
- [x] Ability to store and view favorites at http://localhost:3000/favorites/
- [x] Ability to see "Public Favorites"
![image](https://user-images.githubusercontent.com/6496862/46900843-1c30ad80-ce5e-11e8-91a6-e193ff10d189.png)
- [x] Ability to login and get re-directed back to the page that sent the request
- [x] Ability to detect whether logged in and update the UI by looking at `app.locals.username` (for ex: updating the navbar links)
![image](https://user-images.githubusercontent.com/6496862/46900850-2b176000-ce5e-11e8-8a12-aadde76e1c3f.png)
![image](https://user-images.githubusercontent.com/6496862/46900855-32d70480-ce5e-11e8-8f39-6856b05cf6c4.png)


**What I’m stuck on**
- [ ] Nesting favorites inside of users. I feel weird about doing that — from a UX perspective, it feels odd to expose user ID in the URL like keto-eating.io/users/24/favorites. I think it would be nicer to have a nested route like keto-eating.io/dashboard/favorites — What do you think? Do you agree? Any other  ideas?
- [ ] Requiring authentication for the POST method of saving a favorite. I tried to create a helper class but when I tried to use it, it said `ReferenceError: app is not defined`.
    - To test it:
        - uncomment line 2 on ./controller/favorite.js: ```// const auth = require('./helpers/auth')```
        - Change POST method from ```app.post('/favorites/', (req, res) =>``` to ```app.post('/favorites/', auth.requireLogin,(req, res) =>```
- [ ] (Bonus) Deduplicating favorites that were favorited by many users on the list of “public favorites” at /favorites/ 

@PhyllisWong , do you want to merge this and then open a new branch, or do you want to keep working on this same branch and see if you can finish it before we merge?
